test-utils/state-viewer/src/cli.rs

```diff
@@ -104,13 +101,35 @@ impl StateViewerSubCommand {
 
 #[derive(Clap)]
 pub struct DumpStateCmd {
+    /// Optionally, can specify at which height to dump state.
     #[clap(long)]
     height: Option<BlockHeight>,
+    /// Dumps state records and genesis config into separate files.
+    /// Has reasonable RAM requirements.
+    /// Use for chains with large state, such as mainnet and testnet.
+    #[clap(long)]
+    stream: bool,
+    /// Produces a single json file containing the state.
+    /// Needs the whole state to fit in RAM.
+    /// Use for chains with small state, such as betanet, guildnet, localnet.
+    #[clap(long)]
```

Comment:
nit: I thing you can say something like `#[clap(long, conflicts_with(stream))]` or something like that to make them mutually exclusive.

Reply:
@matklad Thanks, that's what I was looking for, for my own personal scripts.

Reply:
@nikurt @pmnoxx as a meta note, Clap is one of those crates where the answer to "How do I ..." is "Yes". It can do almost anything. Given that we have a fair amount of CLI here and there, it might make sense to dedicate some time to reading the docs: https://docs.rs/clap/latest/clap/struct.Arg.html